### DISCUSSION TITLE
Running on an Ubuntu ARM VM

### DISCUSSION THREAD

**Discussion by ptorab-me** [dangowrt/owrt-ubi-installer#149] [Q&A]
I'm running an Ubuntu 22.04.3 LTS VM on a MacBook M1 (using Parallels); Script fails here, which I think is maybe due to architecture mismatch (x86). Is this the case? Am I even failing (error is confusing -- first it says "cannot execute binary file: Exec format error" and next line says "Success")? Is there any workaround? My Ubuntu info shows "Parallels International GmbH. Parallels ARM Virtual Machine" for hardware model.

```
/home/ptorab/projects/openwrt/owrt-ubi-installer/openwrt-ib/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_mt7622/linux-5.10.176/scripts/dtc/dtc: line 5: /home/ptorab/projects/openwrt/owrt-ubi-installer/openwrt-ib/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_mt7622/linux-5.10.176/scripts/dtc/../../../../../../staging_dir/host/lib/ld-linux-x86-64.so.2: cannot execute binary file: Exec format error
/home/ptorab/projects/openwrt/owrt-ubi-installer/openwrt-ib/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_mt7622/linux-5.10.176/scripts/dtc/dtc: line 5: /home/ptorab/projects/openwrt/owrt-ubi-installer/openwrt-ib/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_mt7622/linux-5.10.176/scripts/dtc/../../../../../../staging_dir/host/lib/ld-linux-x86-64.so.2: Success
+ exit 2
```

If there is no workaround, can I install using files in the official OpenWrt release folder https://downloads.openwrt.org/releases/22.03.5/targets/mediatek/mt7622/

Alternatively, if UBI is important, any chance to update the releases here to latest?
https://github.com/dangowrt/owrt-ubi-installer/releases

**Comment by dangowrt** [comment]
As the OpenWrt ImageBuilder comes with many precompiled binaries intended to run on an x86_64 host, you can try [QEMU User Emulation](https://wiki.debian.org/QemuUserEmulation) to emulate the x86_64 CPU in userspace on your ARMv8 Apple silicon (which will be slow, but should work).
